advanced/Scripts/piholeDebug.sh

```diff
@@ -1281,56 +1281,74 @@ analyze_gravity_list() {
     IFS="$OLD_IFS"
 }
 
+obfuscated_pihole_log() {
+  local pihole_log=("$@")
+  local line
+  local error_to_check_for
+  local line_to_obfuscate
+  local obfuscated_line
+  for line in "${pihole_log[@]}"; do
+      # A common error in the pihole.log is when there is a non-hosts formatted file
+      # that the DNS server is attempting to read.  Since it's not formatted
+      # correctly, there will be an entry for "bad address at line n"
+      # So we can check for that here and highlight it in red so the user can see it easily
+      error_to_check_for=$(echo "${line}" | grep 'bad address at')
+      # Some users may not want to have the domains they visit sent to us
+      # To that end, we check for lines in the log that would contain a domain name
+      line_to_obfuscate=$(echo "${line}" | grep ': query\|: forwarded\|: reply')
+      # If the variable contains a value, it found an error in the log
+      if [[ -n ${error_to_check_for} ]]; then
+          # So we can print it in red to make it visible to the user
+          log_write "   ${CROSS} ${COL_RED}${head_line}${COL_NC} (${FAQ_BAD_ADDRESS})"
```

Comment:
Think I've found a bug...

![image](https://user-images.githubusercontent.com/1998970/124369508-4c4eb900-dc64-11eb-9a52-38d20e4536ed.png)

![image](https://user-images.githubusercontent.com/1998970/124369512-54a6f400-dc64-11eb-9b61-8ca08918a5d4.png)

But it's also in dev, so you didn't introduce it :)

Reply:
https://github.com/pi-hole/pi-hole/pull/4216